homeassistant/components/ezviz/__init__.py

```diff
@@ -1 +1,120 @@
-"""Support for Ezviz devices via Ezviz Cloud API."""
+"""Support for Ezviz camera."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyezviz.client import EzvizClient
+from requests import ConnectTimeout, HTTPError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_REGION, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ATTR_SERIAL,
+    CONF_FFMPEG_ARGUMENTS,
+    DATA_COORDINATOR,
+    DATA_UNDO_UPDATE_LISTENER,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+from .coordinator import EzvizDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=30)
+
+PLATFORMS = ["camera", "sensor", "binary_sensor", "switch"]
+
+
+async def async_setup(hass: HomeAssistantType, config: dict) -> bool:
+    """Set up the Ezviz integration."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up Ezviz from a config entry."""
+
+    if entry.data.get(ATTR_SERIAL):
+        hass.data[DOMAIN][entry.unique_id] = entry.data
```

Comment:
I'd store the entry instead since it has all the entry features and data. If the data attribute is reset we can still use the existing entry instance to get access to the new data.

Reply:
Done.